include/experimental/__p2630_bits/submdspan_mapping.hpp

```diff
@@ -182,6 +182,21 @@ struct deduce_layout_left_submapping<
   }
 };
 
+// We are reusing the same thing for layout_left and layout_left_padded
+// For layout_left as source StaticStride is static_extent(0)
+template<class Extents, size_t NumGaps, size_t StaticStride>
+struct Compute_S_static_layout_left {
+  // Neither StaticStride nor any of the looked for extents can zero.
+  // StaticStride never can be zero, the static_extents we are looking at are associated with 
+  // integral slice specifiers - which wouldn't be valid for zero extent
+  template<size_t ... Idx>
+  MDSPAN_INLINE_FUNCTION
+  static constexpr size_t value(std::index_sequence<Idx...>) {
+    size_t val = ((Idx>0 && Idx<=NumGaps ? (Extents::static_extent(Idx) == dynamic_extent?0:Extents::static_extent(Idx)) : 1) * ... * (StaticStride == dynamic_extent?0:StaticStride));
```

Comment:
```suggestion
    if (StaticStride == dynamic_extent)
      return dynamic_extent; 
    size_t val = StaticStride;
    for (size_t idx: {Idx...})
      if (idx > 0 && idx <= NumGaps) {
        if (Extents::static_extent(idx) == dynamic_extent)
          return dynamic_extent;
        else
          val *= Extents::static_extent(idx);
      }
    return val;
```
would be more readable IMHO.

Reply:
I did a compile time test, and the fold expression is a lot faster 2.8s vs 4.1s on my M2 Mac. 

```c++
#include<cstdint>
#include<initializer_list>
constexpr size_t dynamic_extent = 1;
template<size_t ... Idx>
struct extents {
  constexpr static size_t values[sizeof...(Idx)] = {Idx...};
  constexpr static size_t static_extent(int i) { return values[i]; }
};


template<size_t NumGaps, size_t StaticStride, class Extents, size_t ... Idx>
size_t foo() {
#ifdef FOLD
  size_t val = ((Idx>0 && Idx<=NumGaps ? (Extents::static_extent(Idx) == dynamic_extent?0:Extents::static_extent(Idx)) : 1) * ... * (StaticStride == dynamic_extent?0:StaticStride));
#else
  if (StaticStride == dynamic_extent)
      return dynamic_extent;
    size_t val = StaticStride;
    for (size_t idx: {Idx...})
      if (idx > 0 && idx <= NumGaps) {
        if (Extents::static_extent(idx) == dynamic_extent)
          return dynamic_extent;
        else
          val *= Extents::static_extent(idx);
      }
#endif

  return val;
}


template<size_t ... Idx>
size_t foo_add() {
  if constexpr (sizeof ... (Idx) < 5) {
    return foo_add<Idx ..., 1>() +
    foo_add<Idx ..., 2>() +
    foo_add<Idx ..., 3>() +
    foo_add<Idx ..., 4>() +
    foo_add<Idx ..., 5>() +
    foo_add<Idx ..., 6>() +
    foo_add<Idx ..., 7>() +
    foo_add<Idx ..., 8>() +
    foo_add<Idx ..., 9>();
  } else {
    return foo<1,3,extents<Idx...>, 0, 1, 2, 3, 4>();
  }
}

int main(int, char* argv[]) {
  return foo_add<1>() + foo_add<2>();
}
```